components/script/dom/bindings/callback.rs

```diff
@@ -201,14 +200,8 @@ impl Drop for CallSetup {
         if need_to_deal_with_exception {
             unsafe {
                 let old_global = RootedObject::new(self.cx, self.exception_compartment.ptr);
```

Comment:
You don't need the old_global thing either.  It was only needed because JS_SaveFrameChain nulls out the current compartment, right?

On the other hand, you're setting old_global to something other than the current compartment of cx, so I have no idea what you're really trying to do there...